### PR TITLE
Setup version dependent setup requires

### LINKED ISSUE #6352: make dev version aware setup_requires
We shouldn't include jinja2 and cython in ``setup_requires`` for the released version, but keep them there for development ones. 

We just make sure to set the value based on the version string.

Once it's done remove them for conda-forge, too https://github.com/conda-forge/astropy-feedstock/issues/13
(though that PR has to be done for the new release, not before)

### PR DESCRIPTION
**Pull request by bsipocz**
This should fix #6352

### CODE REVIEW

**Comment by bsipocz** [comment]
@astrofrog - ahh, sure. I did it quite naively s didn't go back to the issue to reread the comments. Will update this to check for the ``.git``.

**Comment by bsipocz** [comment]
@astrofrog - Hmm, checking only for ``.git`` is also not a good enough as the github zip files don't include either the ``.git`` directory but also cythonised c files.  So as Erik suggested the best test would be to check for the sdist specific ``PKG-INFO``, otherwise one needs cython to get a working installation out of the github release, or dev version.

File: setup.py [hunk 1]
```diff
@@ -62,8 +58,11 @@
     'wcslint = astropy.wcs.wcslint:main',
 ]
 
-setup_requires = ['numpy>=' + astropy.__minimum_numpy_version__,
-                  'cython>=0.21', 'jinja2>=2.7']
+setup_requires = ['numpy>=' + astropy.__minimum_numpy_version__]
+
+if not os.path.exists(os.path.join(os.path.dirname(__file__), 'PKG-INFO')):
```
Comment (astrofrog) [inline setup.py]: Maybe add a comment to explain why we chose to do it this way

Comment (bsipocz) [inline setup.py]: done

**Comment by bsipocz** [comment]
@astrofrog - Is this good to go now?

File: setup.py [hunk 2]
```diff
@@ -1,20 +1,16 @@
 #!/usr/bin/env python
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-import glob
 import os
 import sys
+import glob
```
Comment (astrofrog) [inline setup.py]: I approve of this

**Pull request merged**